src/open_samus_returns_rando/files/templates/randomizerpowerup.lua

```diff
@@ -96,6 +101,24 @@ function RandomizerPowerup.MarkLocationCollected(actorOrName)
     Blackboard.SetProp(playerSection, propName, "b", true)
 end
 
+function RandomizerPowerup.ActivateSpecialEnergy(name)
+    local cloud = "TG_SpecialEnergyCloud"
+
+    -- Powerups
+    if string.sub(name, 0, 7) == "LE_Powe" then
+        local powerup = string.find(name, "_", 4)
+        local trigger = cloud .. string.sub(name, powerup)
+        if Game.GetEntity(trigger) ~= nil then
+            SpecialEnergyCloud.ActivateSpecialEnergy(trigger)
+        end
+    end
+
+    -- Aeion abilities
+    if string.sub(name, 0, 17) == "LE_SpecialAbility" and Game.GetEntity(name) ~= nil then
```

Comment:
The entity check doesn't make sense here.

It will be only checked if the first part is `true` (which is) `string.sub(name, 0, 17) == "LE_SpecialAbility"`. In the second part of the `and` you can already be sure that the name starts with that and we wouldn't have been able to pick it up if the entity for the name doesn't exist.
You could check if the `cloud` exists but afaik `cloud` exists for every pickup starting with `LE_SpecialAbility"`.

Reply:
I said the "entity check" not the whole check :p.
Now, you are activating it even when you just pick up any minor item location.
The name check needs to stay.

Reply:
brain moment